packages/web-components/src/components/masthead/masthead-global-bar.ts

```diff
@@ -20,17 +25,73 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  * @element dds-masthead-global-bar
  */
 @customElement(`${ddsPrefix}-masthead-global-bar`)
-class DDSMastheadGlobalBar extends FocusMixin(LitElement) {
+class DDSMastheadGlobalBar extends FocusMixin(HostListenerMixin(LitElement)) {
+  /**
+   * Media query listener handler.
+   */
+  private _mqHandler;
+
   /**
    * The shadow slot this action bar should be in.
    */
   @property({ reflect: true })
   slot = 'profile';
 
+  /**
+   * Search bar opened flag.
+   */
+  @internalProperty()
+  private _openedSearch = false;
+
+  /**
+   * Medium breakpoint flag.
+   */
+  @internalProperty()
+  private _responsiveMd = false;
+
+  /**
+   * Handles toggle event from the search component.
+   *
+   * @param event The event.
+   */
+  @HostListener(`parentRoot:${DDSMastheadSearch.eventToggle}`)
```

Comment:
To support alternate (typically) component composed in masthead, this codebase makes event name configurable in receiving component side ([an example](https://github.com/carbon-design-system/carbon-web-components/blob/v1.5.0/src/components/ui-shell/side-nav.ts#L198-L200)). So please define `eventToggle` static property to each classes receiving the event:

```suggestion
  @HostListener(`parentRoot:eventToggle`)
```